dspace-api/src/main/java/org/dspace/statistics/util/SpiderDetectorServiceImpl.java

```diff
@@ -185,24 +185,27 @@ private void loadPatterns(String directory, List<Pattern> patternList) {
         File patternsDir = new File(spidersDir, directory);
         if (patternsDir.exists() && patternsDir.isDirectory()) {
             for (File file : patternsDir.listFiles()) {
```

Comment:
Have you considered writing a simple FileFilter implementation, passing it into listFiles() here and below?  That would give us one copy of the test, instead of two that must be updated together if our filtering requirements change.

Reply:
Hi mwoodiupui - a FileFilter is a good idea. Squeezing it into this patch seems too ugly. I am thinking about implementing in Eclipse and starting another PR later. What would be the right package for the FileFilter? org.dspace.statistics.util? Or even org.dspace.util?

Reply:
Your choice (below) seems reasonable.  I had thought of just nesting it inside SpiderDetectorServiceImpl, but your solution is easier to reuse.